baseframe/templates/baseframe/mui/forms.html.jinja2

```diff
@@ -101,13 +101,26 @@
   {%- endif %}
 {%- endmacro %}
 
-{% macro rendersubmit(buttons=[], css_class="", tabindex='', cancel_url='', style='horiz', csrf_error=false) -%}
+{% macro rendersubmit(form, ref_id, buttons=[], css_class="", tabindex='', cancel_url='', style='horiz', csrf_error=false) -%}
+{% if form.recaptcha %}
+<script>
+function onInvisibleRecaptchaSubmit() {
+  event.preventDefault();
+  if (grecaptcha.getResponse() === '') {
+    grecaptcha.execute();
+  } else {
+    document.getElementById("{{ ref_id }}").submit();
+  }
+}
+</script>
```

Comment:
This function can be added to `widgetscripts` macro and form, ref_id need not be included as parameters in  rendersubmit